### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by tresorsawasawa**
Below is a list of tasks we completed in this project:
### Group task
- Create Item class in a separate .rb file.
- All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() method is false
 - Create a main.rb file that will serve as your console app entry-point.
 - Implement startup actions:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

### Team member #1: @Gegardus 
- Create a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Label class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
        - should override the method from the parent class
        - should return true if the parent's method returns true OR if cover_state equals "bad"
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
    - List all books
    - List all labels (e.g. 'Gift', 'New')
    - Add a book
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - books table (add all properties and associations from the parent Item class as table columns)
    - labels table

### Team member #2: @tresorsawasawa 
- Create MusicAlbum class in a separate .rb file.
- Create a Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Genre class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the MusicAlbum class
        - should override the method from the parent class
        - should return true if the parent's method returns true AND if on_spotify equals true
        - otherwise, it should return false
 - Add unit tests for all implemented methods.
 - The following options should be available:
    - List all music albums
    - List all genres (e.g 'Comedy', 'Thriller')
    - Add a music album
 - All data should be preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - music_albums table (add all properties and associations from the parent Item class as table columns)
    - genres table

### Team member #3: @fobadara 
- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if the parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
    - List of games
    - List all authors (e.g. 'Stephen King')
    - Add a game
 - All data should be preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - games table (add all properties and associations from the parent Item class as table columns)
    - authors table

### Team member #4: @ahmedtaa 
- Create Movie class in a separate .rb file.
- Create a Source class with an association to the Item class (in a separate .rb file).
- All Movie class properties visible in the diagram should be defined and set up in the constructor method.
- All Source class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Source class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Movie class
        - should override the method from the parent class
        - should return true if the parent's method returns true OR if silent equals true
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
    - List all movies
    - List all sources (e.g. 'From a friend', 'Online shop')
    - Add a movie
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - movies table (add all properties and associations from the parent Item class as table columns)
    - sources table

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative './item'
+
+class Book < Item
+  attr_accessor :title, :publisher, :cover_state
+
```
Comment (ahmetbozaci) [inline classes/book.rb:5]: ### OPTIONAL

- [ ] For best practice, if you don't use attr_writer just use attr_reader. I see you mostly use attr_accessor in all classes.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @tresorsawasawa @fobadara @ahmedtaa @Gegardus  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your presentation video is professional. Please next time make sure the video's length is max 5 minutes.✔️
- App is working as the requirements requested✔️
- Your Github flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: README.md [hunk 2]
```diff
@@ -1 +1,240 @@
...
+<!-- About the project -->
+<!-- [App screenshot](https://example.com) -->
+
+## Description 🏗️
+
+A console app that will help you to keep a record of different types of things you won: books, music albums, movies, and games. Everything will be based on the UML class diagram presented below
+
+![schema](https://raw.githubusercontent.com/microverseinc/curriculum-ruby/main/group-capstone/images/catalog_of_my_things.png?token=GHSAT0AAAAAABQQDFYMY5Z2J6QEWFCWDVOQYTL3YUQ)
+
+<p align="right">(<a href="#top">back to top</a>)</p>
```
[the remark this answers is not in the document]
Comment (tresorsawasawa) [reply README.md:62]: Thank you. Sorry for the error.

File: classes/book.rb (hunk 1 above)
Comment (tresorsawasawa) [reply classes/book.rb:5]: Thank you

**Comment by tresorsawasawa** [comment]
Hey @ahmetbozaci 

Thank You for the review :pray: 
we really appreciate :+1: :+1:

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gegardus** [comment]
Thank you @mikethreels

**Pull request merged**